### PR TITLE
Create shopping list line item by sku

### LINKED ISSUE #1608: Add shopping list line item by sku

### PR DESCRIPTION
**Pull request by katmatt**
Closes #1608

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraft.java [hunk 1]
```diff
@@ -17,13 +17,17 @@
 @JsonDeserialize(as = LineItemDraftDsl.class)
 @ResourceDraftValue(
         abstractBuilderClass = true,
-        factoryMethods = {@FactoryMethod(parameterNames = {"productId"})})
+        factoryMethods = {
+                @FactoryMethod(methodName = "ofSku", parameterNames = {"sku", "quantity"}),
```
Comment (acbeni) [inline commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraft.java]: i didn't understand why the quantity attribute was added for the factory method 🤔

File: commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraftBuilder.java [hunk 2]
```diff
@@ -25,4 +27,26 @@
...
     }
+
+    /**
+     * Creates a new object initialized with the given values.
+     *
+     * @param variantIdentifier initial value for the {@link io.sphere.sdk.carts.LineItemDraft#getProductId()} and {@link io.sphere.sdk.carts.LineItemDraft#getVariantId()} properties
+     * @param quantity          initial value for the {@link io.sphere.sdk.carts.LineItemDraft#getQuantity()} property
+     * @return new object initialized with the given values
+     */
+    public static LineItemDraftBuilder ofVariantIdentifier(final ByIdVariantIdentifier variantIdentifier, @Nullable final Long quantity) {
```
Comment (acbeni) [inline commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraftBuilder.java]: Does it make sense to add the quantity attribute here?

File: commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraftBuilder.java [hunk 3]
```diff
@@ -25,4 +27,26 @@
...
+    }
+
+    /**
+     * Creates a new object initialized with the given values.
+     *
+     * @param skuVariantIdentifier initial value for the {@link io.sphere.sdk.carts.LineItemDraft#getSku()} property
+     * @param quantity             initial value for the {@link io.sphere.sdk.carts.LineItemDraft#getQuantity()} property
+     * @return new object initialized with the given values
+     */
+    public static LineItemDraftBuilder ofSkuVariantIdentifier(final BySkuVariantIdentifier skuVariantIdentifier, @Nullable final Long quantity) {
```
Comment (acbeni) [inline commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraftBuilder.java]: same as above

File: commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraftBuilder.java (hunk 2 above)
Comment (katmatt) [reply commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraftBuilder.java]: This also follows the pattern of `carts/LineItemDraftBuilder`.

File: commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraftBuilder.java (hunk 3 above)
Comment (katmatt) [reply commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraftBuilder.java]: Same as above ;-)

File: commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraftBuilder.java (hunk 2 above)
Comment (acbeni) [reply commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItemDraftBuilder.java]: Cool 😄  now i see its a consistency issue.
Thanks for the explanation 👍

**Pull request merged**